### PR TITLE
CC-1164 Passenger 5 recipe improvements

### PR DESCRIPTION
**Pull request by radamanthus**
## Description of your patch

Adds an option for passenger_monitor to terminate a bloated Passenger worker with kill -9 if it does not respond to kill -USR1 after a grace period

Fixes passenger_monitor so that it loads the worker_memory_size limit from metadata, to allow Support to tweak memory limits

## Recommended Release Notes

Adds an option for passenger_monitor to terminate a bloated Passenger worker with kill -9 if it does not respond to kill -USR1 after a grace period

Fixes passenger_monitor so that it loads the worker_memory_size limit from metadata, to allow Support to tweak memory limits

## Estimated risk

Medium

## Components involved

passenger5 main chef recipe

## Description of testing done

See QA instructions

## QA Instructions

Test on a Passenger environment for the todo application
Boot the environment on the latest v5 stack
Verify that the chef run was successful
As root, run `crontab -l | grep "passenger_monitor todo"`. The output should be:

```
* * * * * /engineyard/bin/passenger_monitor todo -l 800 >/dev/null 2>&1
```

Upgrade to the QA stack
Verify that there are no chef errors
Run `crontab -l | grep passenger_monitor` again. The output should still be the same.
At the app deployment level, add metadata settings for worker_memory_size and passenger_grace_time:

{
  "worker_memory_size": "1473",
  "passenger_grace_time": "182"
}
Click Apply, then run `crontab -l | grep passenger_monitor` again. The output should be:

```
* * * * * /engineyard/bin/passenger_monitor todo -l 1473 -w 182 >/dev/null 2>&1
```

Verify that the only the crontab for the todo application was modified; the crontab entry for the other application should still be at the default:

```
* * * * * /engineyard/bin/passenger_monitor otherapp -l 800 >/dev/null 2>&1
```

Before wrapping up, do a final test to check if there were any passenger worker terminations while you were testing. Run `grep illing /var/log/syslog`. There shouldn't be any results.

### CODE REVIEW

File: cookbooks/passenger5/recipes/monitoring.rb [hunk 1]
```diff
@@ -70,13 +70,14 @@
...
     # leading to performance problems.
+    grace_time = metadata_app_get_with_default(app_name, :passenger_grace_time, 30)
     cron "passenger_monitor_#{app_name}" do
       minute '*'
       hour '*'
       day '*'
       weekday '*'
       month '*'
-      command "/engineyard/bin/passenger_monitor #{app_name} -l #{max_megabytes} >/dev/null 2>&1"
+      command "/engineyard/bin/passenger_monitor #{app_name} -l #{max_megabytes} -w #{grace_time} >/dev/null 2>&1"
```
Comment (crigor) [inline cookbooks/passenger5/recipes/monitoring.rb]: The grace time passed here is not used on the script.

Comment (radamanthus) [inline cookbooks/passenger5/recipes/monitoring.rb]: Thanks. I updated passenger_monitor:

https://github.com/engineyard/ey-cookbooks-stable-v5/blob/ee35a8c9d6680c85cb0544dde66836ee2cc74201/cookbooks/passenger5/files/default/passenger_monitor#L34-L36

**Comment by crigor** [comment]
LGTM

**Comment by crigor** [comment]
New version LGTM.

**Pull request merged**